external/llvm-project/mlir/lib/ExecutionEngine/RocmRuntimeWrappers.cpp

```diff
@@ -38,19 +38,18 @@ class ScopedContext {
 public:
   ScopedContext() {
     // Static reference to HIP primary context for device ordinal defaultDevice.
-    static hipCtx_t context = [] {
-      HIP_REPORT_IF_ERROR(hipInit(/*flags=*/0));
-      hipDevice_t device;
-      HIP_REPORT_IF_ERROR(hipDeviceGet(&device, /*ordinal=*/defaultDevice));
-      hipCtx_t ctx;
-      HIP_REPORT_IF_ERROR(hipDevicePrimaryCtxRetain(&ctx, device));
-      return ctx;
-    }();
-
-    HIP_REPORT_IF_ERROR(hipCtxPushCurrent(context));
+    HIP_REPORT_IF_ERROR(hipInit(/*flags=*/0));
+    hipDevice_t device;
+    HIP_REPORT_IF_ERROR(hipDeviceGet(&device, /*ordinal=*/defaultDevice));
+    hipCtx_t ctx;
+    HIP_REPORT_IF_ERROR(hipDevicePrimaryCtxRetain(&ctx, device));
   }
 
-  ~ScopedContext() { HIP_REPORT_IF_ERROR(hipCtxPopCurrent(nullptr)); }
+  ~ScopedContext() {
+    hipDevice_t device;
+    HIP_REPORT_IF_ERROR(hipDeviceGet(&device, /*ordinal=*/defaultDevice));
+    HIP_REPORT_IF_ERROR(hipDevicePrimaryCtxRelease(device));
+  }
```

Comment:
Honestly, the fix here is to completely drop the ScopedContext stuff and write in the commit message that it's all deprecated.